src/Lecture3.hs

```diff
@@ -210,11 +237,24 @@ types that can have such an instance.
   implement instances! No spoilers :)
 -}
 
--- instance Foldable Weekday where
--- instance Foldable Gold where
--- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+instance Foldable List1 where
+  foldMap :: Monoid m => (a -> m) -> List1 a -> m
+  foldMap fm (List1 x0 []) = fm x0
+  foldMap fm (List1 x0 (x1:xs)) = fm x0 <> foldMap fm (List1 x1 xs)
+
+  foldr :: (a -> b -> b) -> b -> List1 a -> b
+  foldr f xd (List1 x0 []) = f x0 xd
+  foldr f xd (List1 x0 (x1:xs)) = f x0 (foldr f xd (List1 x1 xs))
```

Comment:
You don't need to allocate a new `List1` on every new recursive call and you can reuse the `Foldable` instance for `[]` (and the `foldr` implementation specifically):

```suggestion
  foldr f xd (List1 x xs) = f x (foldr f xd xs)
```